src/Formik.tsx

```diff
@@ -209,25 +213,21 @@ export function useFormik<Values = object>({
   const runSingleFieldLevelValidation = React.useCallback(
     (field: string, value: void | string): Promise<string> => {
       return new Promise(resolve =>
-        resolve(fields.current[field].validate(value))
-      ).then(x => x, e => e);
+        resolve(fieldRegistry.current[field].validate(value))
+      );
     },
-    [fields]
+    [fieldRegistry]
```

Comment:
It's a general recommendation not to use refs in dependencies because their reference never changes. Not that it would cause any harm in this situation, just pointing out... Unless eslint plugin gets crazy about it (👎).